### PR TITLE
Crud users auth

### PR DESCRIPTION
**Pull request by przyjacielpkp**

### CODE REVIEW

File: src/repository/projects.rs [hunk 1]
```diff
@@ -58,7 +58,13 @@ impl ProjectRepository for PgProjectRepository {
...
             JOIN users as u
             ON p.owner_id = u.user_id
-            WHERE p.owner_id = $1
+            LEFT JOIN (
+                SELECT project_id, friend_id 
+                FROM sharing
+                WHERE friend_id = $1
+            ) as shared
+            ON p.project_id= shared.project_id
+            WHERE p.owner_id = $1 or shared.friend_id = $1
```
Comment (ghost) [inline src/repository/projects.rs]:
```suggestion
            LEFT JOIN (
                SELECT project_id, friend_id 
                FROM sharing
                WHERE friend_id = $1
            ) AS shared
            ON p.project_id= shared.project_id
            WHERE p.owner_id = $1 OR shared.friend_id = $1 
```

File: src/repository/sharing.rs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                error!(%err);
+                return Err(ProjectSharingCreateError::Unknown);
+            }
+        };
+        info!("Transaction acquired");
+
+        let check_if_project_exists = "
+            SELECT project_id
+            FROM projects
+            WHERE projects.owner_id= $1 and projects.project_id = $2
```
Comment (ghost) [inline src/repository/sharing.rs]:
```suggestion
            WHERE projects.owner_id = $1 and projects.project_id = $2 
```

File: src/repository/sharing.rs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+use tracing::{error, info, warn};
+
+use crate::domain::crud::CrudInt;
+
+pub enum ProjectSharingCreateError {
+    Unknown,
+}
+pub enum ProjectSharingUpdateError {
+    Unknown,
+}
```
Comment (ghost) [inline src/repository/sharing.rs:15]: I don't require it to change, but mark as TODO

**Review by ghost — APPROVED** [review]

File: src/repository/sharing.rs (hunk 3 above)
Comment (przyjacielpkp) [reply src/repository/sharing.rs:15]: ok

File: src/repository/sharing.rs (hunk 2 above)
Comment (przyjacielpkp) [reply src/repository/sharing.rs]: sure

File: src/repository/projects.rs (hunk 1 above)
Comment (przyjacielpkp) [reply src/repository/projects.rs]: sure thing

**Pull request merged**